modules/ssm/main.tf

```
@@ -0,0 +1,7 @@
+resource "aws_ssm_parameter" "secret_parameters" {
+  for_each = var.secrets
+
+  name  = "/${var.env_namespace}/${each.key}"
```

it would be good to add random string to the secret name because once we tear down the infra, the secret does not actually get deleted right away, which means that if we try to recreate the infra, it'll fail as the secret name already exists

```
resource "random_string" "service_secret_random_suffix" {
  length  = 6
  special = false
}
```

```suggestion
  name  = "/${var.env_namespace}/${each.key}-${random_string.service_secret_random_suffix.result}"
```

---

That's really good to know, thanks Alex. Updated in [5904760](https://github.com/nvminhtue/tyson_devops_ic/pull/25/commits/59047609860edf98baabcee82af918bc90a2f866)